blocks/amp-story/block-selector.js

```diff
@@ -0,0 +1,97 @@
+
+const { __, sprintf } = wp.i18n;
+const { Component } = wp.element;
+const { Button } = wp.components;
+const {
+	dispatch,
+	select
+} = wp.data;
+const {
+	getBlock,
+	isBlockSelected,
+	hasSelectedInnerBlock,
+	getSelectedBlock
+} = select( 'core/editor' );
+const {
+	selectBlock
+} = dispatch( 'core/editor' );
+
+import LayerInserter from './layer-inserter';
+
+class BlockSelector extends Component {
+	render() {
+		if ( ! this.props.rootClientId ) {
+			return null;
+		}
+
+		const rootBlock = getBlock( this.props.rootClientId );
+
+		if ( ! rootBlock.innerBlocks.length ) {
+			return null;
+		}
+
+		let links = [];
+
+		window.lodash.forEachRight( rootBlock.innerBlocks, function( block, index ) {
+			let className = 'component-editor__selector';
+			if ( isBlockSelected( block.clientId ) || hasSelectedInnerBlock( block.clientId ) ) {
+				className += ' is-selected';
+			}
+
+			let title = sprintf( __( 'Layout %d ', 'amp' ), index + 1 );
+			if ( 'amp/amp-story-cta-layer' === block.name ) {
+				title = __( 'CTA Layer', 'amp' );
+			}
+			links.push(
+				<li className={ className } key={ 'selector-' + index }>
+					<Button id={ block.clientId } onClick={ ( e ) => {
+						e.stopPropagation();
+
+						// @todo This selects the first inner child instead for some reason. Note that this also creates a new paragraph as the first child is it doesn't exist.
+						selectBlock( block.clientId );
+
+						// @todo This is a temporary workaround for selecting the correct block. Remove when possible.
+						let timeout = 50;
+						setTimeout( function() {
+							selectBlock( block.clientId );
+						}, timeout );
```

Comment:
Looks like using `selectBlock` a second time with timeout results in selecting the correct block eventually, however, there is another issue as well. Noticed that not the first inner child is selected but the first inner text block is selected. For example if the Grid Layer only has one Image block as the inner child and then `selectBlock` is used then a paragraph block is automatically created within this Grid Layer.

We should definitely look into the issues with `selectBlock` / alternatives at some moment ourselves.